modules/compatibility-tests/src/integrationTest/java/org/apache/ignite/internal/PersistentCompatibilityTest.java

```diff
@@ -0,0 +1,218 @@
+import org.apache.ignite.internal.lang.IgniteSystemProperties;
+import org.apache.ignite.internal.testframework.WithSystemProperty;
+import org.apache.ignite.tx.Transaction;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedClass;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.ValueSource;
+
+/**
+ * Legend: v1 is original version, v2 is current version (version from main branch).
+ * <p>
+ * It is supposed that data is written in v1, then the storage is flushed and the node is stopped, its binary is replaced with v2, the node
+ * is started.
+ * <p>
+ * The tests have to ensure that aipersist-based tables work correctly:
+ * <ul>
+ * <li> All data written in v1 can be read in v2 # including overwritten versions. </li>
+ * <li> Deleted data cannot be read # updates that are rolled back cannot be read. </li>
+ * </ul>
+ * <p>
+ * Variations:
+ * <ul>
+ * <li> When there were no unmerged delta files in v1 (that is, after the writes are finished on
+ * v1, we wait for all delta files to be merged and removed). </li>
+ * <li>When there were some unmerged delta-files. </li>
+ * <li> When unmerged delta-files remain from v1 and we start creating new delta files immediately after starting on v2, and THEN we
+ * start verifications listed above (while the delta files are still there). </li>
+ * </ul>
+ */
+@ParameterizedClass
+@MethodSource("baseVersions")
+@MicronautTest(rebuildContext = true)
+@WithSystemProperty(key = IgniteSystemProperties.THREAD_ASSERTIONS_ENABLED, value = "false")
+public class PersistentCompatibilityTest extends CompatibilityTestBase {
+    private static final String NODE_URL = "http://localhost:" + ClusterConfiguration.DEFAULT_BASE_HTTP_PORT;
+
+    /** Delta files are not compacted before updating the cluster. */
+    private static final String TABLE_WITH_DELTA_FILES = "TEST_WITH_DELTA_FILES";
+
+    /** Delta files are compacted before updating the cluster. */
+    private static final String TABLE_WITHOUT_DELTA_FILES = "TEST_WITHOUT_DELTA_FILES";
+
+    /** Delta files are not compacted before updating the cluster, and new ones created after. */
+    private static final String TABLE_WITH_NEW_DELTA_FILES = "TEST_WITH_NEW_DELTA_FILES";
+
+    @Inject
+    @Client(NODE_URL + "/management/v1/deployment")
+    private HttpClient deploymentClient;
+
+    @Override
+    protected int nodesCount() {
+        return 1;
+    }
+
+    @Override
+    protected void setupBaseVersion(Ignite baseIgnite) {
+        try {
+            deployCheckpointJob();
+
+            createAndPopulateTable(baseIgnite, TABLE_WITHOUT_DELTA_FILES);
+            createAndPopulateTable(baseIgnite, TABLE_WITH_DELTA_FILES);
+            createAndPopulateTable(baseIgnite, TABLE_WITH_NEW_DELTA_FILES);
+
+            // Newly allocated pages are written straight to the page files, we need to avoid this.
```

Comment:
Please improve the comment to make it clear how it relates to the following line